src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2724,3 +2724,8 @@
    )
    
 })
+
+.rs.addFunction("isAbsolutePath", function(path)
+{
+   grepl("^(?:/|~|[a-zA-Z]:[/\\])", path)
```

Comment:
I tried seeing if we could use the Boost utilities here but they don't quite do what we want. In particular, the behavior of `is_absolute()` and `is_relative()` depends on the platform; e.g. on macOS I see:

```
> .Call("rs_isPathAbsolute", "C:/foo/bar")
[1] FALSE
```

But I think it would be better to consider those as absolute paths since they would be considered absolute on Windows.